lib/faraday/http/adapter.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Faraday
+  module Http
+    # HTTP.rb adapter.
+    class Adapter < Faraday::Adapter
+      dependency 'http'
+
+      # Takes the environment and performs the request.
+      #
+      # @param env [Faraday::Env] the request environment.
+      #
+      # @return [Faraday::Response] the response.
+      def call(env)
+        super
+        perform_request(env)
+        @app.call(env)
+      end
+
+      private
+
+      def perform_request(env)
+        conn = setup_connetion(env)
+
+        resp = conn.request env[:method], env[:url], body: env[:body]
+        save_response(env, resp.code, resp.body.to_s, resp.headers, resp.status.reason)
+      rescue ::HTTP::TimeoutError
+        raise Faraday::TimeoutError, $ERROR_INFO
+      rescue ::HTTP::ConnectionError
+        raise Faraday::ConnectionFailed, $ERROR_INFO
+      end
+
+      def setup_connetion(env)
+        conn = ::HTTP
+
+        request_config(conn, env[:request]) if env[:request]
```

Comment:
Is this actually configuring the `conn` object? The var gets replaced inside `#request_config`, but is never updated here in `#setup_connetion`.

```suggestion
        conn = request_config(conn, env[:request]) if env[:request]
```

Reply:
I had to look into the `HTTP.rb` gem, but from what I saw they return a new client every time you call one of these methods 🤦‍♂️
More here: https://github.com/httprb/http/blob/master/lib/http/chainable.rb#L93 (e.g. `timeout`)